### PR TITLE
Add C implementation for `@stdlib/math/base/special/asinh`

### LINKED ISSUE #688: [RFC]: Add C implementation for `@stdlib/math/base/special/asinh`
### Description

This RFC proposes adding C-implementation for [`@stdlib/math/base/special/asinh`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/asinh)
```c
double stdlib_base_asinh( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #688.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks passed, this PR can have a final review.

File: lib/node_modules/@stdlib/math/base/special/asinh/README.md [hunk 1]
```diff
@@ -83,6 +83,94 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/asinh.h"
+#include <stdlib.h>
+#include <stdio.h>
+int main() {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asinh/README.md:154]:
```suggestion

int main() {
```

File: lib/node_modules/@stdlib/math/base/special/asinh/README.md [hunk 2]
```diff
@@ -83,6 +83,94 @@ for ( i = 0; i < x.length; i++ ) {
...
+    double x[] = { -5, -3.89, -2.78, -1.67, -0.55, 0.55, 1.67, 2.78, 3.89, 5 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asinh/README.md:155]:
```suggestion
    double x[] = { -5.0, -3.89, -2.78, -1.67, -0.55, 0.55, 1.67, 2.78, 3.89, 5.0 };
```

File: lib/node_modules/@stdlib/math/base/special/asinh/benchmark/c/native/benchmark.c [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+int main( void ) {
+	double elapsed;
+	int i;
+
+	// Use the current time to seed the random number generator:
+	srand( time( NULL ) );
+
+	print_version();
+	for ( i = 0; i < REPEATS; i++ ) {
+		printf( "# c::%s\n", NAME );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asinh/benchmark/c/native/benchmark.c:130]:
```suggestion
		printf( "# c::native::%s\n", NAME );
```

File: lib/node_modules/@stdlib/math/base/special/asinh/examples/c/example.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/asinh.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x[] = { -5, -3.89, -2.78, -1.67, -0.55, 0.55, 1.67, 2.78, 3.89, 5 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asinh/examples/c/example.c:24]:
```suggestion
    double x[] = { -5.0, -3.89, -2.78, -1.67, -0.55, 0.55, 1.67, 2.78, 3.89, 5.0 };
```

**Review by kgryte — APPROVED** [review]
As the requested changes are minor, I'll merge and fix up on my end. Thanks, @Pranavchiku!

**Pull request merged**